src/sql/workbench/services/notebook/browser/notebookServiceImpl.ts

```diff
@@ -286,6 +288,9 @@ export class NotebookService extends Disposable implements INotebookService {
 		let uri: URI;
 		if (resource) {
 			uri = URI.revive(resource);
+			if (this._notebookInputsMap.has(uri.toString())) {
```

Comment:
I added this input tracker since we call createNotebookInput twice when creating an Interactive notebook. First when creating a notebook document, and again when calling openNotebook. The second input creation doesn't have the initial contents from the first call, so we were losing the Interactive metadata when creating a new notebook. Technically we could loop through the editors to check for this, but a map is faster.

Reply:
Why are we creating multiple inputs? There should only be one input for a given resource typically

Reply:
The VSCode extension APIs allow a user to create a notebook document without opening it an editor, which is the first time we create the input. Our original editor opening code also creates a notebookInput because we formerly only used that way to create new notebooks. It's kind of a compromise so I don't have to redo a bunch of existing editor APIs.

Reply:
Also we need to be able to track notebook inputs separately from editors so that we check those too when creating untitled notebook URIs. Several notebook documents can be created before they're opened in editors.